### PR TITLE
Closes CPPL-271

### PR DESCRIPTION
**Pull request by SantaBar6arra**
- added server support for the change of email

### CODE REVIEW

File: src/Server/Handlers/User/ChangeEmailHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    try {
+        verification = m_facade->GetVerificationCodeByUserId(request.uid);
+    }
+    catch (const db::DatabaseFailure& e) {
+        return FormErrorResponse(
+            NetError::Status::HTTP_INTERNAL_SERVER_ERROR,
+            "Such verification code was not gotten because database error occurred");
+    }
+
+    if (!verification.has_value()) {
```
Comment (sorbeee) [inline src/Server/Handlers/User/ChangeEmailHandler.cpp]: if ( ... )
{
//code
}

File: src/Server/Handlers/User/ChangeEmailHandler.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ChangeEmailHandler();
+    virtual ~ChangeEmailHandler() override {}
+
+    virtual Net::Response AuthHandle(const Net::Request& request);
+private:
+
+    class JSONParser
+    {
+    public:
+        struct DTO
```
Comment (sorbeee) [inline src/Server/Handlers/User/ChangeEmailHandler.h]: I think that parsers need to be moved from handlers

File: src/Server/libdal/Repositories/UserRepository/UserRepository.cpp [hunk 3]
```diff
@@ -112,7 +112,33 @@ bool UserRepository::Update(const User &user)
...
+    return true;
+}
+
+bool UserRepository::UpdateEmail(const IdType user_id, const std::string &email)
+{
+    try
+    {
+        pqxx::work w(m_database_connection);
+
+        auto result = w.exec("SELECT " + user::ID + " FROM  " + user::TABLE_NAME + " WHERE " + user::ID + " = " + w.quote(GetById(user_id)));
```
Comment (sorbeee) [inline src/Server/libdal/Repositories/UserRepository/UserRepository.cpp]: Something wrong with code formatting

File: src/Server/libdal/Repositories/UserRepository/UserRepository.cpp [hunk 4]
```diff
@@ -112,7 +112,33 @@ bool UserRepository::Update(const User &user)
...
+        if (result.empty())
+        {
+            return false;
+        }
+
+        w.exec0(
+            "UPDATE " + user::TABLE_NAME +
+            " SET " +
+                user::EMAIL + " = " + w.quote(email) +
```
Comment (sorbeee) [inline src/Server/libdal/Repositories/UserRepository/UserRepository.cpp]: Something wrong with code formatting too

**Review by sorbeee — CHANGES_REQUESTED** [review]

**Review by sorbeee — APPROVED** [review]

File: src/Server/libdal/Repositories/UserRepository/UserRepository.cpp [hunk 5]
```diff
@@ -117,28 +117,28 @@ bool UserRepository::Update(const User &user)
...
 bool UserRepository::UpdateEmail(const IdType user_id, const std::string &email)
 {
-    try
-    {
-        pqxx::work w(m_database_connection);
+	try
+	{
+		pqxx::work w(m_database_connection);
 
 		auto result = w.exec("SELECT " + user::ID + " FROM  " + user::TABLE_NAME + " WHERE " + user::ID + " = " + w.quote(user_id));
```
Comment (sorbeee) [inline src/Server/libdal/Repositories/UserRepository/UserRepository.cpp:126]: Please write like this:

pqxx::result user_ids = work.exec(
  "SELECT " + user::ID +
  " FROM " + user::TABLE_NAME +
  " WHERE " + user::ID + " = " + work.quote(user_id) + ";");

File: src/Server/libdal/Repositories/UserRepository/UserRepository.cpp [hunk 6]
```diff
@@ -112,6 +112,32 @@ bool UserRepository::Update(const User &user)
...
+		auto result = w.exec("SELECT " + user::ID + " FROM  " + user::TABLE_NAME + " WHERE " + user::ID + " = " + w.quote(user_id));
+		if (result.empty())
+		{
+			return false;
+		}
+
+		w.exec0(
+			"UPDATE " + user::TABLE_NAME +
+			" SET " +
+			user::EMAIL + " = " + w.quote(email) +
```
Comment (yaroslav-kernitskyi) [inline src/Server/libdal/Repositories/UserRepository/UserRepository.cpp]: I think in this case there is not need to move this part to the next line
`" SET " + user::EMAIL + " = " + w.quote(email) +`

File: src/Client/Models/User/ChangeEmailModel.cpp [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#include "ChangeEmailModel.h"
+
+Net::Request ChangeEmailModel::FormRequest(const EmailUpdate& dto, const IdType user_id)
+{
+	Net::Request request;
+	request.uri = m_hostname + "/change-email";
```
Comment (yaroslav-kernitskyi) [inline src/Client/Models/User/ChangeEmailModel.cpp:6]: Why not `/email` and PUT method?

**Review by yaroslav-kernitskyi — APPROVED** [review]

**Pull request merged**